### PR TITLE
Add horizontal gesture support for CupertinoScrollbar

### PR DESCRIPTION
**Pull request by Piinks**
## Description

This updates the gesture support for the CupertinoScrollbar to include scrollables in the horizontal axis. Previously, they only worked for vertical scrollables.

## Related Issues

Part of https://github.com/flutter/flutter/issues/31753

## Tests

CupertinoScrollbar responds to gestures in the horizontal axis.

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by Piinks** [comment]
For more context: I am going to follow similar conventions in updating the material/Scrollbar widget to support gestures. These 'thumbs' will be incorporated into a new DefaultScrollbar that will have a thumb, track, etc.

File: packages/flutter/test/cupertino/scrollbar_test.dart [hunk 1]
```diff
@@ -607,4 +607,77 @@ void main() {
...
+        ),
+      ),
+    );
+
+    expect(scrollController.offset, 0.0);
+
+    // Scroll a bit.
+    const double scrollAmount = 10.0;
+    final TestGesture scrollGesture = await tester.startGesture(tester.getCenter(find.byType(SingleChildScrollView)));
+    // Scroll down by swiping up.
```
Comment (justinmc) [inline packages/flutter/test/cupertino/scrollbar_test.dart:635]: This comment should be updated to "Scroll right by swiping left" if I got that right.

File: packages/flutter/test/cupertino/scrollbar_test.dart [hunk 2]
```diff
@@ -607,4 +607,77 @@ void main() {
...
+    expect(hapticFeedbackCalls, 0);
+    final TestGesture dragScrollbarGesture = await tester.startGesture(const Offset(50.0, 596.0));
+    await tester.pump(_kLongPressDuration);
+    expect(hapticFeedbackCalls, 0);
+    await tester.pump(_kScrollbarResizeDuration);
+    // Allow the haptic feedback some slack.
+    await tester.pump(const Duration(milliseconds: 1));
+    expect(hapticFeedbackCalls, 1);
+
+    // Drag the thumb down to scroll down.
```
Comment (justinmc) [inline packages/flutter/test/cupertino/scrollbar_test.dart:665]: Another comment to update to horizontal.

**Review by justinmc — APPROVED** [review]
LGTM

Just two places where I think comments need to be updated.

File: packages/flutter/test/cupertino/scrollbar_test.dart (hunk 2 above)
Comment (Piinks) [reply packages/flutter/test/cupertino/scrollbar_test.dart:665]: Ah thank you!

File: packages/flutter/test/cupertino/scrollbar_test.dart (hunk 1 above)
Comment (Piinks) [inline packages/flutter/test/cupertino/scrollbar_test.dart:635]:
```suggestion
    // Scroll right by swiping left.
```

File: packages/flutter/test/cupertino/scrollbar_test.dart (hunk 2 above)
Comment (Piinks) [inline packages/flutter/test/cupertino/scrollbar_test.dart:665]:
```suggestion
    // Drag the thumb down to scroll back to the left.
```

**Pull request merged**